app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -94,24 +94,31 @@ class RecentSyncedTabFeature(
     }
 
     private suspend fun dispatchSyncedTabs() {
-        val syncedTab = storage.getSyncedDeviceTabs()
+        val syncedTabs = storage.getSyncedDeviceTabs()
             .filterNot { it.device.isCurrentDevice || it.tabs.isEmpty() }
-            .maxByOrNull { it.device.lastAccessTime ?: 0 }
+            .maxByOrNull { it.tabs.maxOf { tab -> tab.lastUsed } }
             ?.let {
-                val tab = it.tabs.firstOrNull()?.active() ?: return
-                RecentSyncedTab(
-                    deviceDisplayName = it.device.displayName,
-                    deviceType = it.device.deviceType,
-                    title = tab.title,
-                    url = tab.url,
-                    iconUrl = tab.iconUrl
-                )
+                val tabs =
+                    it.tabs.sortedByDescending { tab -> tab.lastUsed }.take(MAX_RECENT_SYNCED_TABS)
```

Comment:
> The tabs should already be in the desired sortedByDescending based on the lastUsed property.

I am okay with this here, but in general I think it can be dangerous to assume implementation details of an API. If `getSyncedDeviceTabs` were even named something like `getSyncedDeviceTabsDescending` or something then I think it would be safe, but IMO the API does not communicate that it sorts anything so the behavior should not be relied on.

Reply:
That's a fair point, we could update the docs to say that they should be in descending. I think that's a reasonable expectation to have from this feature.